am-lib/src/test/java/integration/uk/gov/hmcts/reform/amlib/GrantAccessIntegrationTest.java

```diff
@@ -65,4 +79,26 @@ void whenCreatingResourceForMultipleUsersShouldAppearInDatabase() {
         assertThat(databaseHelper.findExplicitPermissions(resourceId)).hasSize(2)
             .extracting(ExplicitAccessRecord::getAccessorId).containsOnly("y", "z");
     }
+
+    @Test
+    void whenCreatingResourceWithInvalidRelationshipShouldThrowPersistenceException() {
+        Map<JsonPointer, Set<Permission>> multipleAttributePermissions = ImmutableMap.of(
```

Comment:
Why not to use `createPermissionsForWholeDocument`?

Reply:
👍